api/http/server.go

```diff
@@ -732,15 +737,33 @@ func (s *Server) HandleGetFeedRaw(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	data, err := s.api.RetrieveFeedUpdate(r.Context(), ref)
+	timeout := r.URL.Query().Get("timeout")
+	if timeout == "" {
+		timeout = "0"
+	}
+	timeoutDuration, err := time.ParseDuration(fmt.Sprintf("%sms", timeout))
 	if err != nil {
-		httpStatus := http.StatusNotFound
-		respondError(w, r, fmt.Sprintf("feed chunk not found: %s", err), httpStatus)
+		httpStatus := http.StatusBadRequest
+		respondError(w, r, fmt.Sprintf("chunk retrieval fail: %s", err), httpStatus)
 		return
 	}
-	w.Header().Set("Content-Type", api.MimeOctetStream)
-	w.WriteHeader(http.StatusOK)
-	http.ServeContent(w, r, "", time.Now(), bytes.NewReader(data))
+	until := time.Now().Local().Add(timeoutDuration)
+	for {
+		data, err := s.api.RetrieveFeedUpdate(r.Context(), ref)
```

Comment:
I would recommend that we don't do multiple polls here. That should be the responsibility of the lower later. Please consider using time.Timer instead and select on channels instead.

Reply:
I understand it's not ideal to poll here but this was the most straightforward replacement of the client side polling.

If I understand correctly you suggest to create a function in `API`, for example `RetrieveFeedUpdateWithTimeout` that uses a timer inside?

Reply:
Hm, well, actually we should use the `context` for this, but I see that it's not passed through to the `Fetcher` via the `NetStore`, which is kind of shitty (or @acud @janos am I reading that wrong?).

If/when the context is connected, the fetchers will automatically be cancelled when the connection is closed, which should be handled on the client side. If the client wants to try again, it issues a new request.